pkg/kubelet/kubelet.go

```diff
@@ -1863,25 +1915,54 @@ func (kl *Kubelet) recordNodeUnschedulableEvent() {
 // Maintains Node.Spec.Unschedulable value from previous run of tryUpdateNodeStatus()
 var oldNodeUnschedulable bool
 
-// tryUpdateNodeStatus tries to update node status to master.
-func (kl *Kubelet) tryUpdateNodeStatus() error {
-	node, err := kl.kubeClient.Nodes().Get(kl.hostname)
-	if err != nil {
-		return fmt.Errorf("error getting node %q: %v", kl.hostname, err)
-	}
-	if node == nil {
-		return fmt.Errorf("no node instance returned for %q", kl.hostname)
+// setNodeStatus fills in the Status fields of the given Node, overwriting
+// any fields that are currently set.
+func (kl *Kubelet) setNodeStatus(node *api.Node) error {
+	// Set addresses for the node.
+	if kl.cloud != nil {
+		instances, ok := kl.cloud.Instances()
+		if !ok {
+			return fmt.Errorf("failed to get instances from cloud provider")
+		}
+		// TODO(roberthbailey): Can we do this without having credentials to talk
+		// to the cloud provider?
+		nodeAddresses, err := instances.NodeAddresses(kl.hostname)
+		if err != nil {
+			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
+		}
+		node.Status.Addresses = nodeAddresses
+	} else {
+		addr := net.ParseIP(kl.hostname)
+		if addr != nil {
+			node.Status.Addresses = []api.NodeAddress{{Type: api.NodeLegacyHostIP, Address: addr.String()}}
```

Comment:
@gmarek - @bgrant0607 said that we had to keep this behavior in until we remove the v1beta1/v1beta2 apis for backwards compatibility. It should be removed once the older API versions are phased out.